oraclepaas/resource_application_container.go

```diff
@@ -0,0 +1,556 @@
+package oraclepaas
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/go-oracle-terraform/application"
+	opcClient "github.com/hashicorp/go-oracle-terraform/client"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceOraclePAASApplicationContainer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOraclePAASApplicationContainerCreate,
+		Read:   resourceOraclePAASApplicationContainerRead,
+		Delete: resourceOraclePAASApplicationContainerDelete,
+		Update: resourceOraclePAASApplicationContainerUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"manifest_file": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"manifest_attributes"},
+			},
+			"manifest_attributes": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				ConflictsWith: []string{"manifest_file"},
+				MaxItems:      1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"runtime": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"major_version": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(application.ManifestTypeWorker),
+								string(application.ManifestTypeWeb),
+							}, false),
+							Default: string(application.ManifestTypeWorker),
+						},
+						"command": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"release": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"build": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"commit": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"version": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+						"startup_time": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(10, 600),
+							Default:      30,
```

Comment:
I think the defaults should be removed in the manifest block as I think this could have an unexpected side effect of overriding the values set in the manifest.json in the application zip.